src/theory/sets/theory_sets_type_enumerator.cpp

```diff
@@ -89,7 +89,7 @@ SetEnumerator& SetEnumerator::operator++()
     // get a new element and return it as a singleton set
     Node element = *d_elementEnumerator;
     d_elementsSoFar.push_back(element);
-    d_currentSet = d_nodeManager->mkNode(kind::SINGLETON, element);
+    d_currentSet = d_nodeManager->mkSingleton(element.getType(), element);
```

Comment:
Can you find the proper sort to pass here? It should be available in the enumerator.

Reply:
Done.